### PR TITLE
Add snapd device information to ComputerInfo message

### PR DESCRIPTION
**Pull request by st3v3nmw**
depends on https://github.com/Perfect5th/snap-http/pull/6

### CODE REVIEW

File: landscape/client/snap_utils.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    #
+    # <assertion2-headers>: <value>
+    # <assertion2-headers>: <value>
+    #
+    # signature
+
+    # extract the assertion headers + their signatures as separate assertions
+    sections = response.result.decode().split("\n\n")
+    raw_assertions = [
+        "\n\n".join(sections[i : i + 2]) for i in range(0, len(sections), 2)
```
Comment (Perfect5th) [inline landscape/client/snap_utils.py:38]: This split/re-join could be restructured to be more immediately understandable.

Let's think of it as I/O: you have a bytestream with alternating header groups and signatures separated by `'\n\n'`. These are pairs: `(headers, signature)`. Seems like a perfect usecase for `itertools.bulk`. Too bad that's py3.12 only. Here's an alt:

```python3
sections = response.result.decode().split("\n\n")
parsed = []
rest = sections
while rest:
    headers, signature, *rest = rest
    parsed.append(parse_assertion(headers, signature))

return parsed
```

Let me know what you think.

**Review by Perfect5th — CHANGES_REQUESTED** [review]
I think it needs a bit of fixing, in terms of code clarify. See inline comment.

File: landscape/client/snap_utils.py (hunk 1 above)
Comment (st3v3nmw) [reply landscape/client/snap_utils.py]: Updated to use the more readable code with a slight modification to handle cases where there are no assertions retrieved from the API (`response.result = b""` -> `response.result.decode().split("\n\n")
 = [""]` ).

**Review by Perfect5th — APPROVED** [review]
LGTM, thanks!

**Pull request merged**